components/api_server/src/database/measurements.py

```diff
@@ -27,52 +36,24 @@ def count_measurements(database: Database) -> int:
     return int(database.measurements.estimated_document_count())
 
 
-def measurements_by_metric(
-    database: Database,
-    *metric_uuids: MetricId,
-    min_iso_timestamp: str = "",
-    max_iso_timestamp: str = "",
-):
-    """Return recent measurements for the specified metrics, without entities and issue status."""
-    if not metric_uuids:
-        return []
-    measurement_filter: dict = {"metric_uuid": {"$in": metric_uuids}}
-    if min_iso_timestamp:  # pragma: no feature-test-cover
-        measurement_filter["end"] = {"$gt": min_iso_timestamp}
-    if max_iso_timestamp:  # pragma: no feature-test-cover
-        measurement_filter["start"] = {"$lt": max_iso_timestamp}
-    return list(
-        database.measurements.find(
-            measurement_filter,
-            sort=[("start", pymongo.ASCENDING)],
-            projection={
-                "_id": False,
-                "sources.entities": False,
-                "sources.entity_user_data": False,
-                "issue_status": False,
-            },
-        ),
-    )
+def measurements_in_period(database: Database, min_iso_timestamp: str, max_iso_timestamp: str):
+    """Return recent measurements for the specified metrics, without source details and issue status."""
+    measurement_filter = {"start": {"$lte": max_iso_timestamp or iso_timestamp()}, "end": {"$gte": min_iso_timestamp}}
+    return list(database.measurements.find(measurement_filter, sort=START_ASCENDING, projection=NO_MEASUREMENT_DETAILS))
 
 
-def all_metric_measurements(
-    database: Database,
-    metric_uuid: MetricId,
-    max_iso_timestamp: str = "",
-):
+def all_metric_measurements(database: Database, metric_uuid: MetricId, max_iso_timestamp: str):
     """Return all measurements for one metric, without entities and issue status, except for the most recent one."""
-    measurement_filter: dict = {"metric_uuid": metric_uuid}
+    measurement_filter: dict[str, str | dict[str, str]] = {"metric_uuid": str(metric_uuid)}
     if max_iso_timestamp:
-        measurement_filter["start"] = {"$lt": max_iso_timestamp}
-    latest_measurement_complete = database.measurements.find_one(
-        measurement_filter,
-        sort=[("start", pymongo.DESCENDING)],
-        projection={"_id": False},
-    )
-    if not latest_measurement_complete:
+        measurement_filter["start"] = {"$lte": max_iso_timestamp}
+    latest_measurement = database.measurements.find_one(measurement_filter, sort=START_ASCENDING, projection=NO_ID)
+    if not latest_measurement:
         return []
-    all_measurements_stripped = measurements_by_metric(database, metric_uuid, max_iso_timestamp=max_iso_timestamp)
-    return list(all_measurements_stripped)[:-1] + [latest_measurement_complete]
+    all_measurements_stripped = list(
```

Comment:
Would it be a performance improvement to yield a generator instead, i.e. if we don't iterate through the *entire* history? Although that would make more sense in desc sort

Reply:
I don't think it matters, all data is sent to the frontend anyway.